src/overlays/actors/ovl_En_Ru2/z_en_ru2.c

```diff
@@ -575,65 +727,89 @@ void func_80AF3564(EnRu2* this, PlayState* play) {
     }
 }
 
-void func_80AF3604(EnRu2* this, PlayState* play) {
-    func_80AF3564(this, play);
+/**
+ * Action 10
+ */
+void EnRu2_CreditsInvisible(EnRu2* this, PlayState* play) {
+    EnRu2_NextCreditsAction(this, play);
 }
 
-void func_80AF3624(EnRu2* this, PlayState* play) {
-    func_80AF2744(this, play);
+/**
+ * Action 11
+ */
+void EnRu2_CreditsFadeIn(EnRu2* this, PlayState* play) {
+    EnRu2_UpdateBgCheckInfo(this, play);
     EnRu2_UpdateSkelAnime(this);
-    func_80AF2608(this);
-    func_80AF33E0(this);
-    func_80AF34A4(this);
+    EnRu2_UpdateEyes(this);
+    EnRu2_FadeInCredits(this);
+    EnRu2_CheckVisibleInCredits(this);
 }
 
-void func_80AF366C(EnRu2* this, PlayState* play) {
-    func_80AF2744(this, play);
+/**
+ * Action 12
+ */
+void EnRu2_CreditsVisible(EnRu2* this, PlayState* play) {
+    EnRu2_UpdateBgCheckInfo(this, play);
     EnRu2_UpdateSkelAnime(this);
-    func_80AF2608(this);
-    func_80AF3564(this, play);
+    EnRu2_UpdateEyes(this);
+    EnRu2_NextCreditsAction(this, play);
 }
 
-void func_80AF36AC(EnRu2* this, PlayState* play) {
-    s32 something;
+/**
+ * Action 13
+ */
+void EnRu2_CreditsTurnHeadDownLeft(EnRu2* this, PlayState* play) {
+    s32 animDone;
 
-    func_80AF2744(this, play);
-    something = EnRu2_UpdateSkelAnime(this);
-    func_80AF2608(this);
-    func_80AF3530(this, something);
+    EnRu2_UpdateBgCheckInfo(this, play);
+    animDone = EnRu2_UpdateSkelAnime(this);
+    EnRu2_UpdateEyes(this);
+    EnRu2_LookingDownLeft(this, animDone);
 }
 
-void func_80AF36EC(EnRu2* this, PlayState* play) {
-    Flags_SetSwitch(play, func_80AF2690(this));
+/**
+ * Sets the switch indicating that Link met Ruto in the Water Temple.
+ */
+void EnRu2_MarkEncounterOccurred(EnRu2* this, PlayState* play) {
+    Flags_SetSwitch(play, EnRu2_GetSwitchFlag(this));
 }
 
-s32 func_80AF3718(EnRu2* this, PlayState* play) {
-    return Flags_GetSwitch(play, func_80AF2690(this));
+/**
+ * Returns whether Link met Ruto in the Water Temple.
+ */
+s32 EnRu2_EncounterOccurred(EnRu2* this, PlayState* play) {
+    return Flags_GetSwitch(play, EnRu2_GetSwitchFlag(this));
 }
```

Comment:
Similarly, `_GetEncounterSwitchFlag`. Flag accesses in general being named with get and set is pretty helpful imo